src/content_sao.cpp

```diff
@@ -110,6 +110,188 @@ class TestSAO : public ServerActiveObject
 TestSAO proto_TestSAO(NULL, v3f(0,0,0));
 
 /*
+	UnitSAO
+*/
+
+
+UnitSAO::UnitSAO(ServerEnvironment *env, v3f pos) :
+	ServerActiveObject(env, pos),
+	m_hp(-1),
+	m_yaw(0),
+	m_armor_groups_sent(false),
+	m_properties_sent(true),
+	m_animation_range(0,0),
+	m_animation_speed(0),
+	m_animation_blend(0),
+	m_animation_loop(true),
+	m_animation_sent(false),
+	m_bone_position_sent(false),
+	m_attachment_parent_id(0),
+	m_attachment_sent(false)
+{
+	// Initialize something to armor groups
+	m_armor_groups["fleshy"] = 100;
+}
+
+
+void UnitSAO::setArmorGroups(const ItemGroupList &armor_groups)
+{
+	m_armor_groups = armor_groups;
+	m_armor_groups_sent = false;
+}
+
+
+ItemGroupList UnitSAO::getArmorGroups()
+{
+	return m_armor_groups;
+}
+
+
+std::string UnitSAO::getPropertyPacket()
```

Comment:
Is this possible to change this method call to void getPropertyPacket(std::string &result) ? This should avoid a useless copy